pkg/scalers/external_scaler.go

```diff
@@ -197,6 +197,8 @@ func (s *externalScaler) GetMetrics(ctx context.Context, metricName string, metr
 	}
 	defer done()
 
+	// Remove the sX- prefix as the external scaler shouldn't have to know about it
+	metricName = strings.SplitN(metricName, "-", 2)[1]
```

Comment:
Thinking out loud - Might be good to add a unit test for this?

Reply:
+1
and might be actually better to define a function for this functionality next to exisiting one:
https://github.com/kedacore/keda/blob/d9172a74c43ceeafdc235142ddb817a1bc31f974/pkg/scalers/scaler.go#L109

Reply:
Done.